src/output.rs

```diff
@@ -53,32 +54,51 @@ fn print_entry_colorized(
     ls_colors: &LsColors,
     wants_to_quit: &Arc<AtomicBool>,
 ) -> io::Result<()> {
-    let default_style = ansi_term::Style::default();
-
-    // Traverse the path and colorize each component
-    for (component, style) in ls_colors.style_for_path_components(path) {
-        let style = style
-            .map(Style::to_ansi_term_style)
-            .unwrap_or(default_style);
+    // Split the path between the parent and the last component
+    let mut offset = 0;
+    let path_str = path.to_string_lossy();
+
+    if let Some(parent) = path.parent() {
+        offset = parent.to_string_lossy().len();
+        for c in path_str[offset..].chars() {
+            if std::path::is_separator(c) {
+                offset += c.len_utf8();
+            } else {
+                break;
+            }
+        }
```

Comment:
I don't understand how this code could be triggered?

Reply:
Say the (user supplied) path is `a/b///c`.  `parent` will be `a/b`.  This loop then advances the length to include `a/b///`.

Reply:
Ah! Thank you.